libglusterfs, cli, rpc, xlators: consistently use time_t

dmantipov opened the pull request:
Tweak dict and xlator interfaces to allow explicit values of 'time_t' type
and consistently use it for timeouts, time intervals, and whatever similar.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

dmantipov commented:
/run regression

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index 6e230e6ed..159bc931b 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -3325,7 +3325,7 @@ gf_process_reserved_ports(unsigned char *ports, uint32_t ceiling)
 out:
     GF_FREE(ports_info);
 
-#else /* FIXME: Non Linux Host */
+#else  /* FIXME: Non Linux Host */
     ret = 0;
 #endif /* GF_LINUX_HOST_OS */
 
@@ -5185,7 +5185,7 @@ close_fds_except_custom(int *fdv, size_t count, void *prm,
             closer(i, prm);
     }
     sys_closedir(d);
-#else /* !GF_LINUX_HOST_OS */
+#else  /* !GF_LINUX_HOST_OS */
     struct rlimit rl;
     int ret = -1;
 
diff --git a/libglusterfs/src/options.c b/libglusterfs/src/options.c
index d60b38a4c..cc7424668 100644
--- a/libglusterfs/src/options.c
+++ b/libglusterfs/src/options.c
@@ -476,7 +476,8 @@ xlator_option_validate_time(xlator_t *xl, const char *key, const char *value,
 
     if ((input_time < opt->min) || (input_time > opt->max)) {
         snprintf(errstr, 256,
-                 "'%" GF_PRIt "' in 'option %s %s' is "
+                 "'%" GF_PRIt
+                 "' in 'option %s %s' is "
                  "out of range [%.0f - %.0f]",
                  input_time, key, value, opt->min, opt->max);
         gf_smsg(xl->name, GF_LOG_ERROR, 0, LG_MSG_OUT_OF_RANGE, "error=%s",
diff --git a/rpc/rpc-lib/src/rpc-clnt.c b/rpc/rpc-lib/src/rpc-clnt.c
index aee9a26a9..08f97bf2e 100644
--- a/rpc/rpc-lib/src/rpc-clnt.c
+++ b/rpc/rpc-lib/src/rpc-clnt.c
@@ -169,8 +169,8 @@ call_bail(void *data)
 
         gf_log(conn->name, GF_LOG_ERROR,
                "bailing out frame type(%s), op(%s(%d)), xid = 0x%x, "
-               "unique = %" PRIu64 ", sent = %s, timeout = %"
-               GF_PRIt " for %s", trav->rpcreq->prog->progname,
+               "unique = %" PRIu64 ", sent = %s, timeout = %" GF_PRIt " for %s",
+               trav->rpcreq->prog->progname,
                (trav->rpcreq->prog->procnames)
                    ? trav->rpcreq->prog->procnames[trav->rpcreq->procnum]
                    : "--",
diff --git a/xlators/cluster/ec/src/ec.c b/xlators/cluster/ec/src/ec.c
index 247d99cc6..7ee1f7247 100644
--- a/xlators/cluster/ec/src/ec.c
+++ b/xlators/cluster/ec/src/ec.c
@@ -848,8 +848,7 @@ init(xlator_t *this)
     GF_OPTION_INIT("iam-self-heal-daemon", ec->shd.iamshd, bool, failed);
     GF_OPTION_INIT("eager-lock", ec->eager_lock, bool, failed);
     GF_OPTION_INIT("other-eager-lock", ec->other_eager_lock, bool, failed);
-    GF_OPTION_INIT("eager-lock-timeout", ec->eager_lock_timeout, time,
-                   failed);
+    GF_OPTION_INIT("eager-lock-timeout", ec->eager_lock_timeout, time, failed);
     GF_OPTION_INIT("other-eager-lock-timeout", ec->other_eager_lock_timeout,
                    time, failed);
     GF_OPTION_INIT("background-heals", ec->background_heals, uint32, failed);
diff --git a/xlators/mgmt/glusterd/src/glusterd.c b/xlators/mgmt/glusterd/src/glusterd.c
index 001d62894..67c6fe08a 100644
--- a/xlators/mgmt/glusterd/src/glusterd.c
+++ b/xlators/mgmt/glusterd/src/glusterd.c
@@ -935,7 +935,7 @@ out:
     return ret ? -1 : 0;
 }
 #undef RUN_GSYNCD_CMD
-#else /* SYNCDAEMON_COMPILE */
+#else  /* SYNCDAEMON_COMPILE */
 static int
 configure_syncdaemon(glusterd_conf_t *conf)
 {
diff --git a/xlators/protocol/client/src/client.c b/xlators/protocol/client/src/client.c
index f5503e17a..1b1899316 100644
--- a/xlators/protocol/client/src/client.c
+++ b/xlators/protocol/client/src/client.c
@@ -2555,8 +2555,8 @@ reconfigure(xlator_t *this, dict_t *options)
 
     conf = this->private;
 
-    GF_OPTION_RECONF("frame-timeout", conf->rpc_conf.rpc_timeout, options,
-                     time, out);
+    GF_OPTION_RECONF("frame-timeout", conf->rpc_conf.rpc_timeout, options, time,
+                     out);
 
     GF_OPTION_RECONF("ping-timeout", rpc_config.ping_timeout, options, time,
                      out);
diff --git a/xlators/storage/posix/src/posix-common.c b/xlators/storage/posix/src/posix-common.c
index 6100cbe14..b20a5b4bf 100644
--- a/xlators/storage/posix/src/posix-common.c
+++ b/xlators/storage/posix/src/posix-common.c
@@ -1139,8 +1139,8 @@ posix_init(xlator_t *this)
     _private->health_check_active = _gf_false;
     GF_OPTION_INIT("health-check-interval", _private->health_check_interval,
                    time, out);
-    GF_OPTION_INIT("health-check-timeout", _private->health_check_timeout,
-                   time, out);
+    GF_OPTION_INIT("health-check-timeout", _private->health_check_timeout, time,
+                   out);
     if (_private->health_check_interval) {
         ret = posix_spawn_health_check_thread(this);
         if (ret)
diff --git a/xlators/storage/posix/src/posix-helpers.c b/xlators/storage/posix/src/posix-helpers.c
index 3e23614c4..7dfd21334 100644
--- a/xlators/storage/posix/src/posix-helpers.c
+++ b/xlators/storage/posix/src/posix-helpers.c
@@ -2084,9 +2084,9 @@ out:
             ret = 0;
         } else {
             gf_event(EVENT_POSIX_HEALTH_CHECK_FAILED,
-                     "op=%s;path=%s;error=%s;brick=%s:%s timeout is %"
-                     GF_PRIt, op, file_path, strerror(op_errno),
-                     priv->hostname, priv->base_path, timeout);
+                     "op=%s;path=%s;error=%s;brick=%s:%s timeout is %" GF_PRIt,
+                     op, file_path, strerror(op_errno), priv->hostname,
+                     priv->base_path, timeout);
         }
     }
     return ret;
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index 6e230e6ed..159bc931b 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -3325,7 +3325,7 @@ gf_process_reserved_ports(unsigned char *ports, uint32_t ceiling)
 out:
     GF_FREE(ports_info);
 
-#else /* FIXME: Non Linux Host */
+#else  /* FIXME: Non Linux Host */
     ret = 0;
 #endif /* GF_LINUX_HOST_OS */
 
@@ -5185,7 +5185,7 @@ close_fds_except_custom(int *fdv, size_t count, void *prm,
             closer(i, prm);
     }
     sys_closedir(d);
-#else /* !GF_LINUX_HOST_OS */
+#else  /* !GF_LINUX_HOST_OS */
     struct rlimit rl;
     int ret = -1;
 
diff --git a/xlators/mgmt/glusterd/src/glusterd.c b/xlators/mgmt/glusterd/src/glusterd.c
index 9786f9396..2d8fae055 100644
--- a/xlators/mgmt/glusterd/src/glusterd.c
+++ b/xlators/mgmt/glusterd/src/glusterd.c
@@ -935,7 +935,7 @@ out:
     return ret ? -1 : 0;
 }
 #undef RUN_GSYNCD_CMD
-#else /* SYNCDAEMON_COMPILE */
+#else  /* SYNCDAEMON_COMPILE */
 static int
 configure_syncdaemon(glusterd_conf_t *conf)
 {
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index d5fca2f56..4a5f31dfb 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -3325,7 +3325,7 @@ gf_process_reserved_ports(unsigned char *ports, uint32_t ceiling)
 out:
     GF_FREE(ports_info);
 
-#else /* FIXME: Non Linux Host */
+#else  /* FIXME: Non Linux Host */
     ret = 0;
 #endif /* GF_LINUX_HOST_OS */
 
@@ -5178,7 +5178,7 @@ close_fds_except_custom(int *fdv, size_t count, void *prm,
             closer(i, prm);
     }
     sys_closedir(d);
-#else /* !GF_LINUX_HOST_OS */
+#else  /* !GF_LINUX_HOST_OS */
     struct rlimit rl;
     int ret = -1;
 
diff --git a/libglusterfs/src/glusterfs/dict.h b/libglusterfs/src/glusterfs/dict.h
index 6fd8e939e..f2ef6ef0b 100644
--- a/libglusterfs/src/glusterfs/dict.h
+++ b/libglusterfs/src/glusterfs/dict.h
@@ -324,9 +324,9 @@ dict_set_uint64(dict_t *this, char *key, uint64_t val);
 #define dict_get_time(dict, key, val) dict_get_int64((dict), (key), (val))
 #define dict_set_time(dict, key, val) dict_set_int64((dict), (key), (val))
 #elif __WORDSIZE == 32
-#define dict_get_time(dict, key, val)                   \
+#define dict_get_time(dict, key, val)                                          \
     dict_get_int32((dict), (key), ((int32_t *)(val)))
-#define dict_set_time(dict, key, val)                   \
+#define dict_set_time(dict, key, val)                                          \
     dict_set_int32((dict), (key), ((int32_t)(val)))
 #else
 #error "unknown word size"
diff --git a/xlators/mgmt/glusterd/src/glusterd.c b/xlators/mgmt/glusterd/src/glusterd.c
index 9786f9396..2d8fae055 100644
--- a/xlators/mgmt/glusterd/src/glusterd.c
+++ b/xlators/mgmt/glusterd/src/glusterd.c
@@ -935,7 +935,7 @@ out:
     return ret ? -1 : 0;
 }
 #undef RUN_GSYNCD_CMD
-#else /* SYNCDAEMON_COMPILE */
+#else  /* SYNCDAEMON_COMPILE */
 static int
 configure_syncdaemon(glusterd_conf_t *conf)
 {
```

gluster-ant commented:
1 test(s) failed 
./tests/bugs/distribute/bug-862967.t

0 test(s) generated core 


5 test(s) needed retry 
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/georep-basic-dr-tarssh-arbiter.t
./tests/basic/afr/gfid-mismatch-resolution-with-cli.t
./tests/bugs/cli/bug-1077682.t
./tests/bugs/distribute/bug-862967.t
https://build.gluster.org/job/gh_centos7-regression/1518/

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index d5fca2f56..4a5f31dfb 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -3325,7 +3325,7 @@ gf_process_reserved_ports(unsigned char *ports, uint32_t ceiling)
 out:
     GF_FREE(ports_info);
 
-#else /* FIXME: Non Linux Host */
+#else  /* FIXME: Non Linux Host */
     ret = 0;
 #endif /* GF_LINUX_HOST_OS */
 
@@ -5178,7 +5178,7 @@ close_fds_except_custom(int *fdv, size_t count, void *prm,
             closer(i, prm);
     }
     sys_closedir(d);
-#else /* !GF_LINUX_HOST_OS */
+#else  /* !GF_LINUX_HOST_OS */
     struct rlimit rl;
     int ret = -1;
 
diff --git a/libglusterfs/src/glusterfs/dict.h b/libglusterfs/src/glusterfs/dict.h
index 6fd8e939e..f2ef6ef0b 100644
--- a/libglusterfs/src/glusterfs/dict.h
+++ b/libglusterfs/src/glusterfs/dict.h
@@ -324,9 +324,9 @@ dict_set_uint64(dict_t *this, char *key, uint64_t val);
 #define dict_get_time(dict, key, val) dict_get_int64((dict), (key), (val))
 #define dict_set_time(dict, key, val) dict_set_int64((dict), (key), (val))
 #elif __WORDSIZE == 32
-#define dict_get_time(dict, key, val)                   \
+#define dict_get_time(dict, key, val)                                          \
     dict_get_int32((dict), (key), ((int32_t *)(val)))
-#define dict_set_time(dict, key, val)                   \
+#define dict_set_time(dict, key, val)                                          \
     dict_set_int32((dict), (key), ((int32_t)(val)))
 #else
 #error "unknown word size"
diff --git a/xlators/mgmt/glusterd/src/glusterd.c b/xlators/mgmt/glusterd/src/glusterd.c
index 9786f9396..2d8fae055 100644
--- a/xlators/mgmt/glusterd/src/glusterd.c
+++ b/xlators/mgmt/glusterd/src/glusterd.c
@@ -935,7 +935,7 @@ out:
     return ret ? -1 : 0;
 }
 #undef RUN_GSYNCD_CMD
-#else /* SYNCDAEMON_COMPILE */
+#else  /* SYNCDAEMON_COMPILE */
 static int
 configure_syncdaemon(glusterd_conf_t *conf)
 {
```

Commit:
libglusterfs, cli, rpc, xlators: consistently use time_t

Tweak dict and xlator interfaces to allow explicit values of 'time_t' type
and consistently use it for timeouts, time intervals, and whatever similar.
Adjust related tests to prefer 0 over -1 for disabled timeouts.

Signed-off-by: Dmitry Antipov <dantipov@cloudlinux.com>
Updates: #1000

amarts reviewed (commented):
LGTM.

Would be good if someone else also reviews as this involves too many files being touched.

mohit84 commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/bugs/distribute/bug-862967.t

0 test(s) generated core 


8 test(s) needed retry 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
./tests/00-geo-rep/01-georep-glusterd-tests.t
./tests/00-geo-rep/georep-basic-tarssh-ec.t
./tests/basic/afr/ta-shd.t
./tests/basic/volume-snapshot.t
./tests/bugs/distribute/bug-862967.t

3 flaky test(s) marked as success even though they failed 
./tests/000-flaky/basic_mount-nfs-auth.t
./tests/000-flaky/bugs_nfs_bug-1116503.t
./tests/000-flaky/glusterd-restart-shd-mux.t
https://build.gluster.org/job/gh_centos7-regression/1536/

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index d5fca2f56..4a5f31dfb 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -3325,7 +3325,7 @@ gf_process_reserved_ports(unsigned char *ports, uint32_t ceiling)
 out:
     GF_FREE(ports_info);
 
-#else /* FIXME: Non Linux Host */
+#else  /* FIXME: Non Linux Host */
     ret = 0;
 #endif /* GF_LINUX_HOST_OS */
 
@@ -5178,7 +5178,7 @@ close_fds_except_custom(int *fdv, size_t count, void *prm,
             closer(i, prm);
     }
     sys_closedir(d);
-#else /* !GF_LINUX_HOST_OS */
+#else  /* !GF_LINUX_HOST_OS */
     struct rlimit rl;
     int ret = -1;
 
diff --git a/libglusterfs/src/glusterfs/dict.h b/libglusterfs/src/glusterfs/dict.h
index 6fd8e939e..f2ef6ef0b 100644
--- a/libglusterfs/src/glusterfs/dict.h
+++ b/libglusterfs/src/glusterfs/dict.h
@@ -324,9 +324,9 @@ dict_set_uint64(dict_t *this, char *key, uint64_t val);
 #define dict_get_time(dict, key, val) dict_get_int64((dict), (key), (val))
 #define dict_set_time(dict, key, val) dict_set_int64((dict), (key), (val))
 #elif __WORDSIZE == 32
-#define dict_get_time(dict, key, val)                   \
+#define dict_get_time(dict, key, val)                                          \
     dict_get_int32((dict), (key), ((int32_t *)(val)))
-#define dict_set_time(dict, key, val)                   \
+#define dict_set_time(dict, key, val)                                          \
     dict_set_int32((dict), (key), ((int32_t)(val)))
 #else
 #error "unknown word size"
diff --git a/xlators/mgmt/glusterd/src/glusterd.c b/xlators/mgmt/glusterd/src/glusterd.c
index 9786f9396..2d8fae055 100644
--- a/xlators/mgmt/glusterd/src/glusterd.c
+++ b/xlators/mgmt/glusterd/src/glusterd.c
@@ -935,7 +935,7 @@ out:
     return ret ? -1 : 0;
 }
 #undef RUN_GSYNCD_CMD
-#else /* SYNCDAEMON_COMPILE */
+#else  /* SYNCDAEMON_COMPILE */
 static int
 configure_syncdaemon(glusterd_conf_t *conf)
 {
```

mohit84 commented:
/run regression

gluster-ant commented:
1 test(s) failed 
./tests/bugs/distribute/bug-862967.t

0 test(s) generated core 


3 test(s) needed retry 
./tests/00-geo-rep/georep-basic-dr-rsync.t
./tests/basic/glusterd/heald.t
./tests/bugs/distribute/bug-862967.t
https://build.gluster.org/job/gh_centos7-regression/1541/

dmantipov commented:
```
# prove -vf tests/bugs/distribute/bug-862967.t 
tests/bugs/distribute/bug-862967.t .. 
1..8
ok   1 [    117/   1417] <  24> 'glusterd'
ok   2 [     14/     16] <  25> 'pidof glusterd'
ok   3 [     21/    190] <  27> 'gluster --mode=script --wignore volume create patchy localhost.localdomain:/d/backends/patchy0 localhost.localdomain:/d/backends/patchy1 localhost.localdomain:/d/backends/patchy2'
ok   4 [     27/    156] <  28> 'gluster --mode=script --wignore volume set patchy stat-prefetch off'
ok   5 [     23/    290] <  29> 'gluster --mode=script --wignore volume start patchy'
ok   6 [     25/     39] <  32> 'glusterfs --attribute-timeout=0 --entry-timeout=0 --gid-timeout=-1 -s localhost.localdomain --volfile-id patchy /mnt/glusterfs/0'
OK
ok   7 [    103/    179] <  47> 'gluster --mode=script --wignore volume start patchy force'
total 0
ok   8 [  10057/      7] <  57> '0 uid_gid_compare 36 36 36 36'
ok
All tests successful.
Files=1, Tests=8, 13 wallclock secs ( 0.02 usr  0.00 sys +  0.53 cusr  0.45 csys =  1.00 CPU)
Result: PASS
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index d5fca2f56..4a5f31dfb 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -3325,7 +3325,7 @@ gf_process_reserved_ports(unsigned char *ports, uint32_t ceiling)
 out:
     GF_FREE(ports_info);
 
-#else /* FIXME: Non Linux Host */
+#else  /* FIXME: Non Linux Host */
     ret = 0;
 #endif /* GF_LINUX_HOST_OS */
 
@@ -5178,7 +5178,7 @@ close_fds_except_custom(int *fdv, size_t count, void *prm,
             closer(i, prm);
     }
     sys_closedir(d);
-#else /* !GF_LINUX_HOST_OS */
+#else  /* !GF_LINUX_HOST_OS */
     struct rlimit rl;
     int ret = -1;
 
diff --git a/libglusterfs/src/glusterfs/dict.h b/libglusterfs/src/glusterfs/dict.h
index 6fd8e939e..f2ef6ef0b 100644
--- a/libglusterfs/src/glusterfs/dict.h
+++ b/libglusterfs/src/glusterfs/dict.h
@@ -324,9 +324,9 @@ dict_set_uint64(dict_t *this, char *key, uint64_t val);
 #define dict_get_time(dict, key, val) dict_get_int64((dict), (key), (val))
 #define dict_set_time(dict, key, val) dict_set_int64((dict), (key), (val))
 #elif __WORDSIZE == 32
-#define dict_get_time(dict, key, val)                   \
+#define dict_get_time(dict, key, val)                                          \
     dict_get_int32((dict), (key), ((int32_t *)(val)))
-#define dict_set_time(dict, key, val)                   \
+#define dict_set_time(dict, key, val)                                          \
     dict_set_int32((dict), (key), ((int32_t)(val)))
 #else
 #error "unknown word size"
diff --git a/xlators/mgmt/glusterd/src/glusterd.c b/xlators/mgmt/glusterd/src/glusterd.c
index 9786f9396..2d8fae055 100644
--- a/xlators/mgmt/glusterd/src/glusterd.c
+++ b/xlators/mgmt/glusterd/src/glusterd.c
@@ -935,7 +935,7 @@ out:
     return ret ? -1 : 0;
 }
 #undef RUN_GSYNCD_CMD
-#else /* SYNCDAEMON_COMPILE */
+#else  /* SYNCDAEMON_COMPILE */
 static int
 configure_syncdaemon(glusterd_conf_t *conf)
 {
```

dmantipov commented:
/run regression
